packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -308,6 +371,86 @@ export class DatabaseCluster extends DatabaseClusterBase {
     this.singleUserRotationApplication = props.engine.singleUserRotationApplication;
     this.multiUserRotationApplication = props.engine.multiUserRotationApplication;
 
+    let s3ImportRole = props.s3ImportRole;
+    if (props.s3ImportBuckets && props.s3ImportBuckets.length > 0) {
+      if (props.s3ImportRole) {
+        throw new Error(`Only one of s3ImportRole or s3ImportBuckets must be specified, not both.`);
+      }
+
+      s3ImportRole = new Role(this, "S3ImportRole", {
+        assumedBy: new ServicePrincipal("rds.amazonaws.com"),
+        inlinePolicies: {
+          ReadS3Files: new PolicyDocument({
+            statements: [
+              new PolicyStatement({
+                  actions: [
+                    's3:ListBucket',
+                    's3:GetObject',
+                    's3:GetObjectVersion'
+                  ],
+                  resources: bucketAndObjectArns(props.s3ImportBuckets)
+              })
+            ]
+          }),
+        }
+      });
```

Comment:
Suggestion: use `grantRead()`?

```suggestion
      s3ImportRole = new Role(this, "S3ImportRole", {
        assumedBy: new ServicePrincipal("rds.amazonaws.com")
      });
      for (const bucket of props.s3ImportBuckets) {
        bucket.grantRead(s3ImportRole);
      }
```

The logic in `bucketAndObjectArns` is already implemented in `grantRead()`

Reply:
Thanks, that does clean things up!